spring-boot/admin/src/main/java/eu/coatrack/admin/service/GatewayHealthMonitorService.java

```diff
@@ -0,0 +1,132 @@
+ * Each time the user updates a switch or adds a new proxy, this service class is responsible
+ * to process this data and send it to the Front End on the Gateway Health Monitor
+ */
+@Service
+public class GatewayHealthMonitorService {
+
+    @Value("${ygg.gateway-health-monitor.warning.threshold.minutes}")
+    private int gatewayHealthWarningThresholdInMinutes;
+
+    @Value("${ygg.gateway-health-monitor.critical.threshold.minutes}")
+    private int gatewayHealthCriticalThresholdInMinutes;
+
+    @Autowired
+    ProxyRepository proxyRepository;
+
+    public class GatewayDataForTheGatewayHealthMonitor {
+        public String gatewayId;
+        public String name;
+        public ProxyStates status;
+        public Long minutesPastSinceLastContact;
+        public boolean isMonitoringEnabled;
+
+        public void setDataNeededFromProxy(Proxy proxy) {
+            this.gatewayId = proxy.getId();
+            this.name = proxy.getName();
+            this.isMonitoringEnabled = proxy.isMonitoringEnabled();
+        }
+    }
+
+    public ProxyStates gatewayHealthStatusSummary(List<GatewayDataForTheGatewayHealthMonitor> gatewayDataForTheGatewayHealthMonitorList) {
+        List<GatewayDataForTheGatewayHealthMonitor> gatewaysWithMonitoringActivatedList = gatewayDataForTheGatewayHealthMonitorList
+                .stream()
+                .filter(gateway -> gateway.isMonitoringEnabled == true)
+                .collect(Collectors.toList());
+        if (gatewaysWithMonitoringActivatedList
+                .stream()
+                .anyMatch(gateway -> gateway.status == ProxyStates.CRITICAL)) {
+            return ProxyStates.CRITICAL;
+        } else if (gatewaysWithMonitoringActivatedList
+                .stream()
+                .anyMatch(gateway -> gateway.status == ProxyStates.WARNING)) {
+            return ProxyStates.WARNING;
+        } else if (gatewaysWithMonitoringActivatedList.stream().allMatch(gateway -> gateway.status == ProxyStates.OK)) {
+            return ProxyStates.OK;
+        }
+        return ProxyStates.NEVER_CONNECTED;
+    }
+
+    public List<GatewayDataForTheGatewayHealthMonitor> updateProxyInfoForGatewayHealthMonitor() {
+        List<GatewayDataForTheGatewayHealthMonitor> gatewayDataForGatewayHealthMonitorList = new ArrayList<>();
+        List<Proxy> allProxiesOwnedByTheLoggedInUser = proxyRepository.findAvailable();
+        allProxiesOwnedByTheLoggedInUser.forEach((proxy) -> {
+            GatewayDataForTheGatewayHealthMonitor gatewayDataForGatewayHealthMonitor = new GatewayDataForTheGatewayHealthMonitor();
+            gatewayDataForGatewayHealthMonitor.setDataNeededFromProxy(proxy);
+            if (proxy.isMonitoringEnabled()) {
+                if (proxy.getTimeOfLastSuccessfulCallToAdmin() != null) {
+                    gatewayDataForGatewayHealthMonitor.minutesPastSinceLastContact = Duration.between(proxy.getTimeOfLastSuccessfulCallToAdmin(), LocalDateTime.now()).toMinutes();
```

Comment:
Maybe also move this into the constructor of the VO.

Reply:
Done